### PR TITLE
WIP - Added flow for adding OSDs to existing OSD node

### PR DESCRIPTION
**Pull request by shtripat**
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

File: tendrl/commons/objects/definition/master.yaml [hunk 1]
```diff
@@ -71,6 +71,24 @@ namespace.tendrl:
       type: Create
       uuid: 2f94a48a-05d7-408c-b400-e27827f4edef
       version: 1
+    AddOsds:
```
Comment (r0h4n) [inline tendrl/commons/objects/definition/master.yaml]: This flow belongs to integrations.ceph.flows in the node agent (https://github.com/Tendrl/node-agent/tree/develop/tendrl/integrations/ceph)

Anything under the ceph_help files are actually intended to be integration specific flows under NS "integrations.$sds_name"  in node-agent

File: tendrl/commons/flows/create_cluster/ceph_help.py [hunk 2]
```diff
@@ -161,6 +163,40 @@ def create_osds(parameters, created_mons):
...
+                        journal_details[journal_disk_name]['journal_count'] += 1
+                        journal_details[journal_disk_name]['ssd'] = True
+                    else:
+                        journal_details[journal_disk_name] = {
+                            'journal_count': 1,
+                            'ssd': False,
+                            'journal_size': config['journal_size'] * 1024 * 1024
+                        }
+
+            NS.etcd_orm.client.write(
```
Comment (r0h4n) [inline tendrl/commons/flows/create_cluster/ceph_help.py]: - Avoid using the raw client writes, the reasons for this have been widely discussed in the past
- This is sds specific data and belongs to /clusters/$integration_id/ via a new object called integrations.ceph.objects.Journal (attrs: osd_uuid [0]... other journal attributes)
- also create integrations.ceph.objects.Osd, instances of which are created during ceph osd create.

[0]: http://ceph.com/geen-categorie/ceph-osd-uuid-conversion-to-osd-id-and-vice-versa/

File: tendrl/commons/objects/definition/master.yaml (hunk 1 above)
Comment (shtripat) [inline tendrl/commons/objects/definition/master.yaml]: Ok. will move it there.

File: tendrl/commons/flows/create_cluster/ceph_help.py (hunk 2 above)
Comment (shtripat) [inline tendrl/commons/flows/create_cluster/ceph_help.py]: >> This is sds specific data and belongs to /clusters/$integration_id/ via a new object called
>> integrations.ceph.objects.Journal (attrs: osd_uuid 0... other journal attributes)

Will add a class for the same

>> also create integrations.ceph.objects.Osd, instances of which are created during ceph osd
>> create.

Do you mean to save OSD details separately as /clusters/{fsid}/osds other than maps?

Comment (r0h4n) [inline tendrl/commons/flows/create_cluster/ceph_help.py]: > Do you mean to save OSD details separately as /clusters/{fsid}/osds other than maps?
Yes
btw it is not {fsid}, it's /clusters/:integration_id/Osds/:osd_uuid/attributes here... 


> Will add a class for the same. Also this data I need to save only for reference while journal mapping calculation to make sure SSDs are considered if not already 4 journals created on them. Actual OSD jorunal details would be available under maps with partition path etc.

Create the integrations.ceph.objects.Journal  and include journal detail from maps and also store the data that you require in this PR

Comment (shtripat) [inline tendrl/commons/flows/create_cluster/ceph_help.py]: May be as part of another PR?

**Comment by shtripat** [comment]
Closing as add osd flow is moved to node-agent integrations
